src/analysis/processing/qgsalgorithmpointstolines.cpp

```diff
@@ -0,0 +1,226 @@
+/***************************************************************************
+                         qgsalgorithmdpointstolines.cpp
+                         ---------------------
+    begin                : November 2020
+    copyright            : (C) 2020 by Stefanos Natsis
+    email                : uclaros at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpointstolines.h"
+#include "qgsvectorlayer.h"
+
+///@cond PRIVATE
+
+QString QgsPointsToLinesAlgorithm::name() const
+{
+  return QStringLiteral( "pointstolines" );
+}
+
+QString QgsPointsToLinesAlgorithm::displayName() const
+{
+  return QObject::tr( "Points to lines" );
+}
+
+QString QgsPointsToLinesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm takes a point layer and connects its features creating a new line layer.\n\n"
+                      "An attribute or expression may be specified to define the order the points should be connected. "
+                      "If no order expression is specified, the fid is used.\n\n"
+                      "A natural sort can be used when sorting by a string attribute "
+                      "or expression (ie. place 'a9' before 'a10').\n\n"
+                      "An attribute can be selected to group points having the same value into the same resulting line." );
+}
+
+QStringList QgsPointsToLinesAlgorithm::tags() const
+{
+  return QObject::tr( "create,lines,points,connect,convert,join" ).split( ',' );
+}
+
+QString QgsPointsToLinesAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsPointsToLinesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+void QgsPointsToLinesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "CLOSE_LINES" ),
+                QObject::tr( "Create closed lines" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "ORDER_EXPRESSION" ),
```

Comment:
Please add the older ORDER_FIELD compatibility code here from PointsToPaths.py also